File: packages/backend/src/lib/ids.ts
import { randomUUID } from 'node:crypto';

export function newSessionId(): string {
  return randomUUID();
}

export function newRequestId(): string {
  return `SR${Date.now().toString(36).toUpperCase()}${Math.floor(Math.random() * 900 + 100)}`;
}

/**
 * 訂單編號：對齊 mms_order_record.order_no 的可讀格式（YYMMDD + 流水）
 * 例：260801000123
 */
export function newOrderNo(): string {
  const d = new Date();
  const yy = String(d.getFullYear()).slice(2);
  const mm = String(d.getMonth() + 1).padStart(2, '0');
  const dd = String(d.getDate()).padStart(2, '0');
  const seq = String(Math.floor(Math.random() * 1_000_000)).padStart(6, '0');
  return `${yy}${mm}${dd}${seq}`;
}

export function nowIso(): string {
  return new Date().toISOString();
}

/** 回傳 n 天後的 YYYY-MM-DD */
export function isoDatePlus(days: number): string {
  const d = new Date();
  d.setDate(d.getDate() + days);
  return d.toISOString().slice(0, 10);
}

File: packages/api/op_agent/ids.py
"""ID 與時間工具。"""

from __future__ import annotations

import random
import uuid
from datetime import date, datetime, timedelta, timezone


def new_session_id() -> str:
    return str(uuid.uuid4())


def new_request_id() -> str:
    stamp = format(int(datetime.now(timezone.utc).timestamp() * 1000), "x").upper()
    return f"SR{stamp}{random.randint(100, 999)}"


def new_order_no() -> str:
    """訂單編號：對齊 mms_order_record.order_no 的可讀格式（YYMMDD + 6 碼流水）。"""
    d = date.today()
    return f"{d.year % 100:02d}{d.month:02d}{d.day:02d}{random.randint(0, 999999):06d}"


def now_iso() -> str:
    return datetime.now(timezone.utc).isoformat()


def iso_date_plus(days: int) -> str:
    """回傳 n 天後的 YYYY-MM-DD。"""
    return (date.today() + timedelta(days=days)).isoformat()


def today_iso() -> str:
    return date.today().isoformat()
